### PR TITLE
[PREVIEW] Pro 3175 f implement new gds design system pattern for check your answers

### PR DESCRIPTION
**Pull request by adamb-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-3175

### Change description ###
Implement the new GDS Design System patterns for Check Your Answers/Summary page.  This is also 
used on the copies summary page.  It now uses a definition list <dl>,<dt> and <dd> instead of tables.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: app/steps/ui/summary/includes/deceased.njk [hunk 1]
```diff
@@ -28,25 +28,20 @@
...
-            <div class="column-one-half">
-            {% for index, otherName in fields.deceased.otherNames.value %}
-                <div class="grid-row">
-                    <div class="column-one-half">{{otherName.firstName.value | safe}} {{otherName.lastName.value | safe}}</div>
-                </div>
-            {% endfor %}
+    <div class="check-your-answers__row">
+        <dt class="check-your-answers__question">{{content.Summary.otherNamesLabel | safe}}</dt>
+        <dd class="check-your-answers__answer">
+        {% for index, otherName in fields.deceased.otherNames.value %}
```
Comment (shepda) [inline app/steps/ui/summary/includes/deceased.njk]: We could remove `index` as it's not being used in the loop

File: app/views/widgets/checkanswer.html [hunk 2]
```diff
@@ -1,35 +1,33 @@
...
-<tr>
-    <td colspan="2">
+
     {% set complete = false  %}
     {% for item in items %}
     {% if item.visible %}
     {% if item.answer%}{% set complete = true  %}{% endif %}
-        <div class="grid-row">
-            <div class="column-one-half">{{item.question | safe}}</div>
+        <div class="check-your-answers__row {{'answer-group' if answerIsGroup}}">
```
Comment (shepda) [inline app/views/widgets/checkanswer.html]: Do we still need `answer-group` as we're now displaying a change link for each line?

File: app/views/widgets/checkanswer.html [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
+            <dt class="check-your-answers__question">{{item.question | safe}}</dt>
             {% if answerIsCollection %}
```
Comment (shepda) [inline app/views/widgets/checkanswer.html]: Do we still need `answerIsCollection` as we're now displaying a change link for each line?

File: app/steps/ui/summary/includes/deceased.njk (hunk 1 above)
Comment (adamb-moj) [reply app/steps/ui/summary/includes/deceased.njk]: Removing index causes the loop to fail.  Leaving it in.

File: app/views/widgets/checkanswer.html (hunk 2 above)
Comment (adamb-moj) [reply app/views/widgets/checkanswer.html]: Done.  Removed answer-group as no longer needed.

File: app/views/widgets/checkanswer.html (hunk 3 above)
Comment (adamb-moj) [reply app/views/widgets/checkanswer.html]: answerIsCollection is still used so has not been removed.

File: app/views/widgets/checkanswer.html [hunk 4]
```diff
@@ -7,12 +7,14 @@
...
-        <div class="check-your-answers__row {{'answer-group' if answerIsGroup}}">
+        <div class="check-your-answers__row">
             <dt class="check-your-answers__question">{{item.question | safe}}</dt>
             {% if answerIsCollection %}
             <dd class="check-your-answers__answer">
                 {% for answ in item.answer %}
-                {{answ | safe}} <br/>
+                    <div class="check-your-answers__row">
+                    {{answ | safe}}
+                    </div>
```
[the remark this answers is not in the document]
Comment (adamb-moj) [reply app/views/widgets/checkanswer.html]: Fixed

**Comment by nunheadblue** [comment]
Tested and passed on Preview - ready to merge

**Pull request merged**